plugin-server/src/cdp/groups-manager.ts

```diff
@@ -0,0 +1,137 @@
+import { Hub, Team } from '../types'
+import { PostgresUse } from '../utils/db/postgres'
+import { GroupType, HogFunctionInvocationGlobals } from './types'
+
+export type GroupsMap = Record<string, GroupType>
+export type GroupsCache = Record<Team['id'], GroupsMap>
+
+// Maps to the group type index for easy lookup like: { 'team_id:group_type': group_type_index }
+type GroupIndexByTeamType = Record<string, number | undefined>
+
+type Group = {
+    id: string
+    index: number
+    type: string
+    url: string
+    properties: Record<string, any>
+    teamId?: number
+}
+
+export class GroupsManager {
+    constructor(private hub: Hub) {}
+
+    private async filterTeamsWithGroups(teams: Team['id'][]): Promise<Team['id'][]> {
+        const teamIds = await Promise.all(
+            teams.map(async (teamId) => {
+                if (await this.hub.organizationManager.hasAvailableFeature(teamId, 'group_analytics')) {
+                    return teamId
+                }
+            })
+        )
+
+        return teamIds.filter((x) => x !== undefined) as Team['id'][]
+    }
+
+    private async fetchGroupTypesMapping(teams: Team['id'][]): Promise<GroupIndexByTeamType> {
+        const teamsWithGroupAnalytics = await this.filterTeamsWithGroups(teams)
+
+        const result = await this.hub.postgres.query(
+            PostgresUse.COMMON_READ,
+            `SELECT team_id, group_type, group_type_index FROM posthog_grouptypemapping WHERE team_id = ANY($1)`,
+            [teamsWithGroupAnalytics],
+            'fetchGroupTypes'
+        )
+
+        return result.rows.reduce(
+            (acc, row) => ({
+                ...acc,
+                [`${row.team_id}:${row.group_type}`]: row.group_type_index,
+            }),
+            {} as GroupIndexByTeamType
+        )
+    }
```

Comment:
This seems like something that'll rarely change. Do we want any kind of LRU cache on it to spare some queries, or is this premature optimization?